api/window/src/main/java/quarks/window/Window.java

```diff
@@ -157,4 +158,10 @@ Licensed to the Apache Software Foundation (ASF) under one
      */
     Function<T, K> getKeyFunction();
 
+    /**
+     * Retrieves the partitions in the window.
+     * @return A map of the window's keys and partitions.
```

Comment:
The returned map is stable when synchronizing on its intrinsic lock, for example:
```java
Map<Integer, Partition<Integer, Integer, ArrayList<Integer> > > partitions = window.getPartitions();
synchronized(partitions){
    // Stable operations on/with partitions
}
```
I've amended the doc to address this.